views/index.jade

```diff
@@ -117,16 +117,73 @@ block content
           h5.modal-title CI Deployment
         .modal-body
           h4 Select the repository
-          form(action="/ci/register", method="POST")
-            if repos !== undefined
-              select.form-control(name="repository")
-                each repo in repos
-                  option #{repo}
-            .modal-footer
-              button.btn.btn-default(type="submit") Register
-  if repos !== undefined
+          form#ci_register(action="/ci/register", method="POST")
+            div.input-group
+              input.form-control#search_bar(placeholder="Enter project name")
+              span.input-group-btn
+                button.btn.btn-default#search(type="button") Search
+            br
+            div#search_result                         
+           .modal-footer
+              button.btn.btn-default.disabled#register_btn(type="submit") Register
+  if ci === true
     script.
       $("#ciModal").modal('show');
+      $('#ci_register').on('keyup keypress', function(e) {
+        var keyCode = e.keyCode || e.which;
+        if (keyCode === 13) { 
+          e.preventDefault();
+          return false;
+        }
+      });
+      let search = () => {
```

Comment:
don't use this! Use basic function definition of JS (Structure for JS in the project should be same.)
```
var search = function () {
  ....
  ....
}
```

Also, can you try and move all this JS to 
`public/scripts/ci.js`